### ISSUE TITLE
Crosshair takes significant time to generate valid examples for some strategies

### ISSUE THREAD

**Issue by tybug** [pschanely/hypothesis-crosshair#16]
I'm seeing many related internal `IgnoreAttempt` raises here.

For instance:
```python
@given(st.datetimes())
@settings(backend="crosshair", deadline=None)
def f(n):
    print("call")
f()
```

takes ~75 `IgnoreAttempts` over 5 seconds until it prints its first `call`.

Other offenders are `st.dates()` (not that bad) and `st.emails()`.

This may not be surprising given crosshair is working with a low level IR, but I wonder if this could be alleviated in some way?

**Comment by pschanely** [comment]
Haven't forgotten about this! Will at least have a diagnostic response this weekend.

**Comment by pschanely** [comment]
Dates involve many bounded integers, so we have to make a bunch of decisions correctly to get a valid date; but there's some internal crosshair stuff I can do to force it to make certain decisions if possible, which helps a lot. Fixed in hypothesis-crosshair v0.0.10
